### PR TITLE
Migrating network visualizations form to react hook form and TS.

### PR DESCRIPTION
**Pull request by AAJELLAL**

### CODE REVIEW

File: src/components/dialogs/parameters/network-visualizations/map-parameters.tsx [hunk 1]
```diff
@@ -36,11 +45,102 @@ export const MapParameters = () => {
             label: '100',
         },
     ];
+    const watchLineFlowColorMode = useWatch({
+        name: `${TabValue.MAP}.${PARAM_LINE_FLOW_COLOR_MODE}`,
+    });
+
+    const onSliderChange = (value: any) => {
```
Comment (YenguiSeddik) [inline src/components/dialogs/parameters/network-visualizations/map-parameters.tsx:52]: I think here you can use number as type: const onSliderChange = (value: number)

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    },
+    {
+        id: LineFlowMode.ANIMATED_ARROWS,
+        label: 'AnimatedArrows',
+    },
+    {
+        id: LineFlowMode.FEEDERS,
+        label: 'Feeders',
+    },
+];
```
Comment (YenguiSeddik) [inline src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:76]: you can use const here :
```suggestion
];export const INTL_LINE_FLOW_MODE_OPTIONS = [
    {
        id: LineFlowMode.STATIC_ARROWS,
        label: 'StaticArrows',
    },
    {
        id: LineFlowMode.ANIMATED_ARROWS,
        label: 'AnimatedArrows',
    },
    {
        id: LineFlowMode.FEEDERS,
        label: 'Feeders',
    },
];
```

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+export const intlLineFlowColorModeOptions = () => [
+    {
+        id: LineFlowColorMode.NOMINAL_VOLTAGE,
+        label: 'NominalVoltage',
+    },
+    {
+        id: LineFlowColorMode.OVERLOADS,
+        label: 'Overloads',
+    },
+];
```
Comment (YenguiSeddik) [inline src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:87]: same here

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    },
+    {
+        id: MAP_BASEMAP_CARTO,
+        label: 'Carto',
+    },
+    {
+        id: MAP_BASEMAP_CARTO_NOLABEL,
+        label: 'CartoNoLabel',
+    },
+];
```
Comment (YenguiSeddik) [inline src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:101]: same here

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+export const intlSubstationLayoutOptions = () => [
+    {
+        id: SubstationLayout.HORIZONTAL,
+        label: 'HorizontalSubstationLayout',
+    },
+    {
+        id: SubstationLayout.VERTICAL,
+        label: 'VerticalSubstationLayout',
+    },
+];
```
Comment (YenguiSeddik) [inline src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:135]: same here

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+export const formatParametersToSend = (
+    parameters: NetworkVisualizationParametersForm,
+    changedParameters: Partial<NetworkVisualizationParametersForm>
+) => {
+    const getFlattenedObject = (nestedObject: NestedObject) =>
+        Object.values(nestedObject)
+            .flatMap((obj) => Object.entries(obj))
+            .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
```
Comment (YenguiSeddik) [inline src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:144]: Here if you are trying to have all the params in forme of a Record<string, string | boolean | number>, i think it should be like this : 
```suggestion
            const getFlattenedObject = (nestedObject: NestedObject): Record<string, string | number | boolean> => {
    return Object.entries(nestedObject).reduce((acc, [key, value]) => {
        if (typeof value === 'object') {
            const flattenedChild = getFlattenedObject(value as NestedObject);
            return { ...acc, ...flattenedChild };
        } else {
            return { ...acc, [key]: value };
        }
    }, {} as Record<string, string | number | boolean>);
};
```

**Review by jamal-khey — COMMENTED** [review]
there is a lot of re-renders every time something changes, it's probably due to some components created using function in map-parameters.tsx

File: src/components/dialogs/parameters/network-visualizations/map-parameters.tsx (hunk 1 above)
Comment (AAJELLAL) [reply src/components/dialogs/parameters/network-visualizations/map-parameters.tsx]: Done

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts (hunk 2 above)
Comment (AAJELLAL) [reply src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:76]: Done

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts (hunk 3 above)
Comment (AAJELLAL) [reply src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:87]: Done

File: src/components/dialogs/parameters/network-visualizations/map-parameters.tsx [hunk 7]
```diff
@@ -36,11 +45,102 @@ export const MapParameters = () => {
...
+        setValue(`${TabValue.MAP}.${PARAM_LINE_FLOW_ALERT_THRESHOLD}`, Number(value), {
+            shouldValidate: true,
+        });
+    };
```
[the remark this answers is not in the document]
Comment (AAJELLAL) [reply src/components/dialogs/parameters/network-visualizations/map-parameters.tsx]: Done

File: src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts (hunk 6 above)
Comment (AAJELLAL) [reply src/components/dialogs/parameters/network-visualizations/network-visualizations-utils.ts:144]: I really don't see why I need to change this. What seems to be the problem with the actual getFlattenedObject?

File: src/components/dialogs/parameters/network-visualizations/map-parameters.tsx [hunk 8]
```diff
@@ -49,23 +49,21 @@ export const MapParameters = () => {
...
     }, []);
 
     const lineFlowColorModeOptions = useMemo(() => {
-        return intlLineFlowColorModeOptions();
+        return INTL_LINE_FLOW_COLOR_MODE_OPTIONS;
     }, []);
 
     const mapBaseMapOptions = useMemo(() => {
-        return intlMapBaseMapOptions();
+        return INTL_MAP_BASE_MAP_OPTIONS;
```
Comment (jamal-khey) [inline src/components/dialogs/parameters/network-visualizations/map-parameters.tsx:66]: useMemo is not usefull when you return a constant

Comment (AAJELLAL) [reply src/components/dialogs/parameters/network-visualizations/map-parameters.tsx]: Indeed. I forgot to remove it

**Review by jamal-khey — APPROVED** [review]
issue with validate button : fixed
issue with component library null : fixed.

Looks good to me.

**Pull request merged**